snippet_ranger/model2/base_decompose.py

```diff
@@ -0,0 +1,69 @@
+from typing import Iterable
+
+from modelforge import Model
+from ast2vec import Model2Base
+
+
+class Model2BaseDecompose(Model2Base):
+    """
+    Class for model entry filtration, decomposition, filtration again and merging.
+    Produce exactly one model from one, but it has more entries because of decomposition.
+    MODEL_FROM should be iterable.
+    """
+    def __init__(self, *args, **kwargs):
```

Comment:
Yes, I can split it more accurate using information from UAST of the library. or example, if I have a big function I can remove all unnecessary code and take code that actually uses the library.
I think It will be more when I get some results and see the snippets.

But It can be inherited from `Source2Func` from the other side.